oci/namespaces.go

```diff
@@ -4,13 +4,9 @@ import specs "github.com/opencontainers/runtime-spec/specs-go"
 
 // RemoveNamespace removes the `nsType` namespace from OCI spec `s`
 func RemoveNamespace(s *specs.Spec, nsType specs.NamespaceType) {
-	idx := -1
 	for i, n := range s.Linux.Namespaces {
 		if n.Type == nsType {
-			idx = i
+			s.Linux.Namespaces = append(s.Linux.Namespaces[:i], s.Linux.Namespaces[i+1:]...)
```

Comment:
Regardless of the original intention, this change increases the chances of a panic.
@thaJeztah : change your go playground code to remove the element `c` instead of `a`.

append may allocate a new list, while the loop still iterate through the original one.